File: src/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {

    public static void main(String[] args) {
        int[] response = selectionSort(new int[]{3, 0, -4, 1, 100});

        System.out.println(Arrays.toString(response));
    }

    static int[] selectionSort(int[] arr) {
        for(int i = 0; i < arr.length - 1; i++) {
            int minIndex = i;

            for(int j = i + 1; j < arr.length; j++) {
                if(arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }

            swap(i, minIndex, arr);
        }

        return arr;
    }

    static void swap(int num1, int num2, int[] arr) {
        int tmp = arr[num1];

        arr[num1] = arr[num2];
        arr[num2] = tmp;
    }


}

File: js/selectionSort.js
// Time Complexity: O(n^2)
// Space Complexity: O(1)
const selectionSort = (array) => {

  for(let i = 0; i < array.length; i++) {
    let minIndex = i;

    for(let j = i + 1; j < array.length; j++) {
      if(array[minIndex] > array[j]) {
        minIndex = j;
      }
    }

    [array[i], array[minIndex]] = [array[minIndex], array[i]];
  }

  return array;
}

console.log(selectionSort([5,4,3,2,1]));
